tests/gis_tests/test_geoip2.py

```diff
@@ -1,50 +1,52 @@
-import os
+import itertools
 import pathlib
 from unittest import mock, skipUnless
 
 from django.conf import settings
 from django.contrib.gis.geoip2 import HAS_GEOIP2
 from django.contrib.gis.geos import GEOSGeometry
-from django.test import SimpleTestCase
+from django.test import SimpleTestCase, override_settings
 from django.utils.deprecation import RemovedInDjango60Warning
 
 if HAS_GEOIP2:
+    import geoip2
+
     from django.contrib.gis.geoip2 import GeoIP2, GeoIP2Exception
 
+GEOIP_PATH = pathlib.Path(__file__).parent.joinpath("data/geoip2").resolve()
```

Comment:
We try to avoid accessing `__file__` at the module level (ticket-32316, 9ee693bd6cf4074f04ec51c6f3cfe87cad392f12).

Reply:
Ah, yes. Thanks for the reminder. Will have to ponder this one as I'm using `@override_settings` at the top level...

Reply:
I've added a fixup that should address this. I didn't want to move `@override_settings` inside every test but I found [`unittest.TestCase.enterClassContext()`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.enterClassContext) which is typically 3.11+ only 💀

In the long run, this will be the cleanest approach and we can drop the backport. If you feel this is a bit clunky to add here we could just implement a streamlined version here for what we require:

```python
    if not PY311:
        @classmethod
        def enterClassContext(cls, cm):
            result = type(cm).__enter__(cm)
            cls.addClassCleanup(type(cm).__exit__, cm, None, None, None)
            return result
```

Reply:
h/t to @adamchainz for his [blog post](https://adamj.eu/tech/2022/11/14/unittest-context-methods-python-3-11-backports/) which I stumbled on 🎩